### PR TITLE
Update adaptive_soft_matching.py

### PR DESCRIPTION
**Pull request by ujjwal-shekhar**
Changed repeated assignment to List comprehension

### CODE REVIEW

File: dipy/denoise/adaptive_soft_matching.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from dipy.core import wavelet
 
 
-def adaptive_soft_matching(ima, fimau, fimao, sigma):
+def adaptive_soft_matching(ima, fimau, fimao, sigma):6
```
Comment (skoudoro) [inline dipy/denoise/adaptive_soft_matching.py:6]: typo: 6 ?

**Review by skoudoro — COMMENTED** [review]
Hi @ujjwal-shekhar,

Apart from the typo, it looks good to me. Thank you for this update.

File: dipy/denoise/adaptive_soft_matching.py (hunk 1 above)
Comment (ujjwal-shekhar) [reply dipy/denoise/adaptive_soft_matching.py]: Hello @skoudoro,
I'm sorry about the typo; I have edited it now.

Also, There were many checks that were failing. Is it an issue with my PR?

**Comment by skoudoro** [comment]
Thank you @ujjwal-shekhar, merging

**Pull request merged**